internal/ls/userpreferences.go

```diff
@@ -209,3 +348,289 @@ func (p *UserPreferences) ModuleSpecifierPreferences() modulespecifiers.UserPref
 		AutoImportSpecifierExcludeRegexes: p.AutoImportSpecifierExcludeRegexes,
 	}
 }
+
+// ------ Parsing Config Response -------
+
+// returns non-nil if should break loop
+func (p *UserPreferences) Parse(item any) *UserPreferences {
+	if item == nil {
+		// continue
+	} else if config, ok := item.(map[string]any); ok {
+		p.parseWorker(config)
+	} else if item, ok := item.(*UserPreferences); ok {
+		// case for fourslash
+		return item.CopyOrDefault()
+	}
+	return nil
+}
+
+func (p *UserPreferences) parseWorker(config map[string]interface{}) {
+	// Process unstable preferences first so that they do not overwrite stable properties
+	if unstable, ok := config["unstable"]; ok {
+		// unstable properties must be named the same as userPreferences
+		p.parseAll(unstable)
+	}
+	for name, values := range config {
+		switch name {
+		case "unstable":
+			continue
+		case "inlayHints":
+			p.parseInlayHints(values)
+		case "suggest":
+			p.parseSuggest(values)
+		case "preferences":
+			p.parsePreferences(values)
+		case "format":
+			// !!!
```

Comment:
Is this challenging to fill out? This is where we are wanting it next, I believe.

Reply:
I'm working on it at the moment, I didn't want to bog this PR down farther